### PR TITLE
Removed authentication domains from CredentialRequest

### PR DESCRIPTION
**Pull request by dxslly**
Code changes to implement Proposal from issue #5 
- Removed authentication domains from the CredentialRequest (Updated protobuff as a non-backwards compatible change to leave a cleaner API.  My rational is that until this API hits v1 breaking changes are okay.)
- Update test app and demo providers to conform to new API changes.
- Updated specification to reflect API changes.
- Fixed Trapdoor demo provider's CredentialQueryReceiver implementation to match it's documentation.

### CODE REVIEW

File: api/proto/openyolo.proto [hunk 1]
```diff
@@ -21,9 +21,8 @@ package org.openyolo;
 option java_package = "org.openyolo.proto";
 
 message CredentialRetrieveRequest {
-    repeated string       authDomains        = 1; // at least one required
-    repeated string       authMethods        = 2; // at least one required
-    repeated KeyValuePair additionalParams   = 3;
+    repeated string       authMethods        = 1; // at least one required
```
Comment (iainmcgin) [inline api/proto/openyolo.proto]: be careful when modifying protos - renumbering fields can horribly break things. Leave the original numbers the same, and put a comment in this file that field 1 is deprecated. It's not super important just now when there are no production apps using openyolo, but we should start treating it like it is.

File: spec/en/retrieving-credentials.md [hunk 2]
```diff
@@ -8,21 +8,14 @@ NOTE: writing this chapter is *in progress*.
 
 ```protobuf
 message CredentialRetrieveRequest {
-    repeated string       authDomains        = 1; // at least one required
-    repeated string       authMethods        = 2; // at least one required
-    repeated KeyValuePair additionalParams   = 3;
+    repeated string       authMethods        = 1; // at least one required
```
Comment (iainmcgin) [inline spec/en/retrieving-credentials.md]: make sure this matches the proto file once you update it

File: api/proto/openyolo.proto (hunk 1 above)
Comment (dxslly) [inline api/proto/openyolo.proto]: Done. Marked as deprecated and added comment (In javadoc style, not sure what the style guide is proto comments).

File: spec/en/retrieving-credentials.md (hunk 2 above)
Comment (dxslly) [inline spec/en/retrieving-credentials.md]: Done.

File: api/javatests/org/openyolo/api/AuthenticationDomainTest.java [hunk 3]
```diff
@@ -60,111 +78,153 @@
...
-        assertEquals(expectedUri, authDomain.getUri().toString());
+
+        assertThat(authDomain.getUri().toString()).isEqualTo(VALID_AUTHENTICATION_DOMAIN_STRING);
     }
 
     @Test(expected = RequireViolation.class)
-    public void testCreateAndroidAuthDomain_nullDomain() throws Exception{
-        AuthenticationDomain.createAndroidAuthDomain(null, new Signature(""));
+    public void CreateAndroidAuthDomain_withNullDomain_throwsRequireViolation() throws Exception{
+        AuthenticationDomain.createAndroidAuthDomain(null /* authDomain */, new Signature(""));
```
Comment (tikurahul) [inline api/javatests/org/openyolo/api/AuthenticationDomainTest.java]: nit: 
```java
new Signature("" /* what the parameter is.*/)
```

File: spi/javatests/java/org/openyolo/spi/BaseCredentialQueryReceiverTest.java [hunk 4]
```diff
@@ -46,87 +47,149 @@
 import org.robolectric.annotation.Config;
 
 /**
- * Battery of tests for BaseCredentialQueryReceiver
+ * Unit tests for {@link BaseCredentialQueryReceiver}
```
Comment (tikurahul) [inline spi/javatests/java/org/openyolo/spi/BaseCredentialQueryReceiverTest.java]: nit: end comments with `.`

File: api/java/org/openyolo/api/RetrieveRequest.java [hunk 5]
```diff
@@ -47,59 +44,53 @@
...
+                .build();
     }
 
     /**
-     * The set of authentication methods that the requestor supports. This is used to filter the
-     * set of credentials saved by the provider. If no authentication methods are specified, then
-     * no filtering of saved credentials will occur.
+     * Creates a {@link RetrieveRequest} from the given authentication methods.
+     */
+    public static RetrieveRequest forAuthenticationMethods(@NonNull Uri... authenticationMethods) {
```
Comment (dxslly) [inline api/java/org/openyolo/api/RetrieveRequest.java]: Ack. The codebase currently uses this style.  See line 181: setAuthenticationMethods(Uri... authMethods).

File: spi/javatests/java/org/openyolo/spi/BaseCredentialQueryReceiverTest.java (hunk 4 above)
Comment (dxslly) [inline spi/javatests/java/org/openyolo/spi/BaseCredentialQueryReceiverTest.java]: Done.

File: api/javatests/org/openyolo/api/AuthenticationDomainTest.java (hunk 3 above)
Comment (dxslly) [inline api/javatests/org/openyolo/api/AuthenticationDomainTest.java]: Done.

File: api/javatests/org/openyolo/api/AuthenticationDomainTest.java [hunk 6]
```diff
@@ -16,41 +16,59 @@
...
 import org.openyolo.api.internal.CollectionConverter;
 import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 import org.valid4j.errors.RequireViolation;
 
-import java.util.ArrayList;
-import java.util.List;
-
+/**
+ * Unit tests for {@link AuthenticationDomain}
```
Comment (dxslly) [inline api/javatests/org/openyolo/api/AuthenticationDomainTest.java]: Done.

**Pull request merged**